optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -537,16 +537,20 @@ def pre_attn_forward(
                     else:
                         kv_seq_len = past_key_value[0].shape[-2]
 
-        if position_embeddings is None:
```

Comment:
this code is coming from upstream https://github.com/huggingface/transformers/blob/v4.43.0/src/transformers/models/llama/modeling_llama.py 

Instead of commenting out this code which looks ugly, similar result should be achieved if you move this check inside the constructor. Can you pls try that?